File: Samples/Layout/Typography/VB.NET/Typography.vb
Imports System.IO
Imports BitMiracle.Docotic.Pdf.Layout

Namespace BitMiracle.Docotic.Pdf.Samples
    NotInheritable Class TypographySample
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Const PathToFile As String = "Typography.pdf"
            PdfDocumentBuilder.Create().Generate(PathToFile,
                Sub(doc)
                    doc.Typography(New CustomTypography(doc))

                    doc.Pages(
                        Sub(page)
                            page.MarginTop(50)

                            page.Header().Border(Sub(b) b.Bottom(1)).Text("Header")

                            page.Content().Text(
                                Sub(text)
                                    text.Line("Regular")

                                    text.Line("Title").Style(Function(t) t.Title)
                                    text.Line("SubTitle").Style(Function(t) t.SubTitle)
                                    text.Line("Heading1").Style(Function(t) t.Heading1)
                                    text.Line("Heading2").Style(Function(t) t.Heading2)
                                    text.Line("Heading3").Style(Function(t) t.Heading3)

                                    text.Line("Caption").Style(Function(t) t.Caption)
                                    text.Line("Footnote").Style(Function(t) t.Footnote)

                                    text.Hyperlink("Hyperlink", New Uri("https://bitmiracle.com"))
                                    text.Line()

                                    text.Line("Emphasis").Style(Function(t) t.Emphasis)
                                    text.Line("Strong").Style(Function(t) t.Strong)
                                    text.Line("printf(""Plain"");").Style(Function(t) t.Plain)

                                    text.Line("Sample").Style(Function(t) TryCast(t, CustomTypography).Sample)
                                End Sub)

                            page.Footer().Border(Sub(b) b.Top(1)).Text(
                                Sub(t)
                                    t.Span("Footer ")
                                    t.CurrentPageNumber()
                                End Sub)
                        End Sub)
                End Sub)

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")
            Process.Start(New ProcessStartInfo(PathToFile) With {.UseShellExecute = True})
        End Sub
    End Class

    Class CustomTypography
        Inherits Typography
        Public Sub New(doc As Document)
            MyBase.New(doc)

            ' Use the custom font with the default document style
            Dim roboto = New FileInfo("..\Sample Data\Fonts\Roboto\Roboto-Regular.ttf")
            Document = doc.TextStyleWithFont(roboto)

            ' Customize the size of the page header font
            Header = Parent.FontSize(8)

            ' Customize the size and style of the page footer font
            Footer = Emphasis.FontSize(8)

            ' Create a custom style
            Sample = Parent.Strikethrough().Underline().FontColor(New PdfRgbColor(255, 0, 0))
        End Sub

        Public Sample As TextStyle
    End Class

End Namespace

File: Samples/Layout/Typography/C#/Typography.cs
using System;
using System.Diagnostics;
using System.IO;
using BitMiracle.Docotic.Pdf.Layout;

namespace BitMiracle.Docotic.Pdf.Samples
{
    static class TypographySample
    {
        static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            const string PathToFile = "Typography.pdf";
            PdfDocumentBuilder.Create().Generate(PathToFile, doc =>
            {
                doc.Typography(new CustomTypography(doc));

                doc.Pages(page =>
                {
                    page.MarginTop(50);

                    page.Header().Border(b => b.Bottom(1)).Text("Header");

                    page.Content().Text(t =>
                    {
                        t.Line("Regular");

                        t.Line("Title").Style(t => t.Title);
                        t.Line("SubTitle").Style(t => t.SubTitle);
                        t.Line("Heading1").Style(t => t.Heading1);
                        t.Line("Heading2").Style(t => t.Heading2);
                        t.Line("Heading3").Style(t => t.Heading3);

                        t.Line("Caption").Style(t => t.Caption);
                        t.Line("Footnote").Style(t => t.Footnote);

                        t.Hyperlink("Hyperlink", new Uri("https://bitmiracle.com"));
                        t.Line();

                        t.Line("Emphasis").Style(t => t.Emphasis);
                        t.Line("Strong").Style(t => t.Strong);
                        t.Line("printf(\"Plain\");").Style(t => t.Plain);

                        t.Line("Sample").Style(t => ((CustomTypography)t).Sample);
                    });

                    page.Footer().Border(b => b.Top(1)).Text(t =>
                    {
                        t.Span("Footer ");
                        t.CurrentPageNumber();
                    });
                });
            });

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");
            Process.Start(new ProcessStartInfo(PathToFile) { UseShellExecute = true });
        }
    }

    class CustomTypography : Typography
    {
        public CustomTypography(Document doc)
            : base(doc)
        {
            // Use custom font in the default document style
            FileInfo roboto = new(@"..\Sample Data\Fonts\Roboto\Roboto-Regular.ttf");
            Document = doc.TextStyleWithFont(roboto);

            // Customize size of the page header font
            Header = Parent.FontSize(8);

            // Customize size and style of the page footer font
            Footer = Emphasis.FontSize(8);

            // Custom style
            Sample = Parent.Strikethrough().Underline().FontColor(new PdfRgbColor(255, 0, 0));
        }

        public TextStyle Sample { get; }
    }
}
